File: frontend/src/types/car.ts
export interface SourceInfo {
  id: number;
  name: string;
  domain: string;
  base_url: string;
  reliability_score: number;
  source_type?: string;
}

export interface CarVariantSummary {
  id: number;
  manufacturer_name: string;
  model_name: string;
  variant_name: string;
  model_year: number;
  body_type: string;
  fuel_type: string;
  transmission: string;
  ex_showroom_price: number;
  estimated_on_road_price: number;
  currency: string;
  combined_mileage?: number;
  electric_range?: number;
  seating_capacity: number;
  airbags: number;
  safety_rating?: number;
  image_url?: string;
  is_saved?: boolean;
}

export interface CarDetail {
  id: number;
  manufacturer_name: string;
  model_name: string;
  variant_name: string;
  model_year: number;
  body_type: string;
  ex_showroom_price: number;
  estimated_on_road_price: number;
  currency: string;
  country: string;
  fuel_type: string;
  transmission: string;
  engine_cc?: number;
  cylinders?: number;
  horsepower?: number;
  torque_nm?: number;
  mileage_city?: number;
  mileage_highway?: number;
  combined_mileage?: number;
  battery_capacity?: number;
  electric_range?: number;
  charging_time?: number;
  seating_capacity: number;
  airbags: number;
  safety_rating?: number;
  boot_space?: number;
  ground_clearance?: number;
  length?: number;
  width?: number;
  height?: number;
  wheelbase?: number;
  drive_type: string;
  features?: Record<string, any>;
  safety_features?: Record<string, any>;
  infotainment_features?: Record<string, any>;
  comfort_features?: Record<string, any>;
  pros?: string[];
  cons?: string[];
  description?: string;
  image_url?: string;
  source_url?: string;
  source?: SourceInfo;
  is_saved?: boolean;
  last_updated: string;
}

export interface CarSearchFilterParams {
  query?: string;
  manufacturer?: string;
  body_type?: string;
  fuel_type?: string;
  transmission?: string;
  price_min?: number;
  price_max?: number;
  min_mileage?: number;
  min_airbags?: number;
  min_safety_rating?: number;
  page?: number;
  page_size?: number;
}

File: backend/app/schemas/car.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, ConfigDict

class SourceResponse(BaseModel):
    id: int
    name: str
    domain: str
    base_url: str
    reliability_score: float
    source_type: str

    model_config = ConfigDict(from_attributes=True)


class CarVariantSummary(BaseModel):
    id: int
    manufacturer_name: str
    model_name: str
    variant_name: str
    model_year: int
    body_type: str
    fuel_type: str
    transmission: str
    ex_showroom_price: float
    estimated_on_road_price: float
    currency: str
    combined_mileage: Optional[float] = None
    electric_range: Optional[float] = None
    seating_capacity: int
    airbags: int
    safety_rating: Optional[float] = None
    image_url: Optional[str] = None
    is_saved: Optional[bool] = False

    model_config = ConfigDict(from_attributes=True)


class CarDetailResponse(BaseModel):
    id: int
    manufacturer_name: str
    model_name: str
    variant_name: str
    model_year: int
    body_type: str
    ex_showroom_price: float
    estimated_on_road_price: float
    currency: str
    country: str
    
    fuel_type: str
    transmission: str
    engine_cc: Optional[int] = None
    cylinders: Optional[int] = None
    horsepower: Optional[float] = None
    torque_nm: Optional[float] = None

    mileage_city: Optional[float] = None
    mileage_highway: Optional[float] = None
    combined_mileage: Optional[float] = None
    battery_capacity: Optional[float] = None
    electric_range: Optional[float] = None
    charging_time: Optional[float] = None

    seating_capacity: int
    airbags: int
    safety_rating: Optional[float] = None
    boot_space: Optional[int] = None
    ground_clearance: Optional[int] = None
    length: Optional[int] = None
    width: Optional[int] = None
    height: Optional[int] = None
    wheelbase: Optional[int] = None
    drive_type: str

    features: Optional[Dict[str, Any]] = None
    safety_features: Optional[Dict[str, Any]] = None
    infotainment_features: Optional[Dict[str, Any]] = None
    comfort_features: Optional[Dict[str, Any]] = None
    pros: Optional[List[str]] = None
    cons: Optional[List[str]] = None
    description: Optional[str] = None
    image_url: Optional[str] = None
    source_url: Optional[str] = None
    source: Optional[SourceResponse] = None
    is_saved: Optional[bool] = False
    last_updated: datetime

    model_config = ConfigDict(from_attributes=True)


class CarSearchFilter(BaseModel):
    query: Optional[str] = None
    manufacturer: Optional[str] = None
    body_type: Optional[str] = None
    fuel_type: Optional[str] = None
    transmission: Optional[str] = None
    price_min: Optional[float] = None
    price_max: Optional[float] = None
    min_mileage: Optional[float] = None
    min_airbags: Optional[int] = None
    min_safety_rating: Optional[float] = None
    seating_capacity: Optional[int] = None
    page: int = 1
    page_size: int = 12


class CarCompareRequest(BaseModel):
    variant_ids: List[int]
